addons/point_of_sale/models/pos_order.py

```diff
@@ -823,7 +823,7 @@ def _order_line_fields(self, line):
     def create(self, values):
         if values.get('order_id') and not values.get('name'):
             # set name based on the sequence specified on the config
-            config_id = self.env['pos.order'].browse(values['order_id']).session_id.config_id.id
+            config_id = self.env[self.order_id._name].browse(values['order_id']).session_id.config_id.id
```

Comment:
I don't like that but it is probably the easiest, add a comment above to explain why we do that